crates/fuel-core/src/state/historical_rocksdb.rs

```diff
@@ -1273,11 +1473,202 @@ mod tests {
         assert!(!historical_rocks_db.is_migration_in_progress());
     }
 
+    #[tokio::test]
+    async fn historical_rocksdb_perform_migration_on_new_instance() {
+        // Given
+        let rocks_db = RocksDb::<Historical<OnChain>>::default_open_temp(None).unwrap();
+
+        let historical_rocks_db =
+            InnerHistoricalRocksDB::new(rocks_db, StateRewindPolicy::RewindFullRange)
+                .unwrap();
+
+        // Commit 100 blocks
+        for i in 1..=100u32 {
+            let mut transaction = historical_rocks_db.read_transaction();
+            transaction
+                .storage_as_mut::<ContractsAssets>()
+                .insert(&key(), &(123 + i as u64))
+                .unwrap();
+            historical_rocks_db
+                .commit_changes(Some(i.into()), transaction.into_changes())
+                .unwrap();
+        }
+
+        let mut revert_migration_transaction = StorageTransaction::transaction(
+            &historical_rocks_db.db,
+            ConflictPolicy::Overwrite,
+            Changes::default(),
+        );
+
```

Comment:
Maybe we can add assert here? It's not crucial, but helps a bit with the clarity of the flow.

```
        let v1_changes = historical_rocks_db
            .db
            .iter_all::<ModificationsHistoryV1<OnChain>>(None)
            .count();
        let v2_changes = historical_rocks_db
            .db
            .iter_all::<ModificationsHistoryV2<OnChain>>(None)
            .count();
        assert_eq!(v1_changes, 0);
        assert_eq!(v2_changes, 100);
```